File: sampleproject/lib/APIs/models/chargingPoint.dart
class chargingPointModel {
  late String name;
  late String lat;
  late String long;
  late String open;
  late String close;
  late String days;
  late String id;

  chargingPointModel(this.name, this.lat, this.long, this.open, this.close,
      this.days, this.id);

  chargingPointModel.fromJson(Map<String, dynamic> map) {
    this.name = map['name'];
    this.lat = map['lat'];
    this.long = map['long'];
    this.open = map['openingTime'];
    this.close = map['closingTime'];
    this.days = map['days'];
    this.id = map['_id'];
  }
}

File: Mongo/src/models/chargingPoint.js
const mongoose = require("mongoose");

const chargingPointsSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  lat: {
    type: String,
    required: true,
  },
  long: {
    type: String,
    required: true,
  },
  openingTime: {
    type: String,
    default: "09:00 صباحا",
  },
  closingTime: {
    type: String,
    default: "05:00 مساءا",
  },
  days: {
    type: String,
    default: "سبت احد اثنين ثلاثاء اربعاء خميس",
  },
});

chargingPointsSchema.methods.toJSON = function () {
  const point = this;
  const pointObject = point.toObject();
  return pointObject;
};

const chargingPoint = mongoose.model("chargingPoint", chargingPointsSchema);
module.exports = chargingPoint;
